File: python/leetcode/Solution002.py
from data import ListNode

class Solution002(object):
    def addTwoNumbers(self, l1, l2):
        if not l1 or not l2:
            return l1 if l1 else l2
        dummy = ListNode(0);
        ptr, carry = dummy, 0
        while l1 or l2 or carry:
            val1 = l1.val if l1 else 0
            val2 = l2.val if l2 else 0
            sums = val1 + val2 + carry
            ptr.next = ListNode(sums % 10)
            ptr = ptr.next
            carry = sums / 10
            l1 = l1.next if l1 else None
            l2 = l2.next if l2 else None
        return dummy.next

File: java/src/leetcode/one/Solution002.java
package JavaAnswer.one;

import JavaAnswer.ListNode;

public class Solution002 {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        if (l1 == null || l2 == null) {
            return l1 == null ? l2 : l1;
        }
        ListNode dummy = new ListNode(0);
        ListNode ptr = dummy;
        int carry = 0;
        while (l1 != null || l2 != null || carry != 0) {
            int valOfl1 = l1 != null ? l1.val : 0;
            int valOfl2 = l2 != null ? l2.val : 0;
            int sum = valOfl1 + valOfl2 + carry;
            ptr.next = new ListNode(sum % 10);
            carry = sum / 10;
            ptr = ptr.next;
            l1 = l1 != null ? l1.next : null;
            l2 = l2 != null ? l2.next : null;
        }
        return dummy.next;
    }
}